### PR TITLE
Refactoring season

### PR DESCRIPTION
**Pull request by AlexisDumortier**
- reduced the number of lines for each methods
- refactored the worst/winningest_coach

### CODE REVIEW

File: lib/season.rb [hunk 1]
```diff
@@ -146,35 +126,25 @@ def regular_postseason_with_percentage(season_id)
...
+    reg_season_perc.reduce({}) do |result, perc|
+      !post_season_perc[perc[0]].nil? ? result[perc[0]] = perc[1] - post_season_perc[perc[0]] : result[perc[0]] = perc[1]
       result
     end
   end
 
   def biggest_bust(season_id)
-    perc_difference = percent_difference(season_id)
-    biggest_bust_team_id = perc_difference.max_by {|team| team[1]}
+    biggest_bust_team_id = percent_difference(season_id).max_by {|team| team[1]}
```
Comment (AlexisDumortier) [inline lib/season.rb]: removed useless lines

File: lib/season.rb [hunk 2]
```diff
@@ -146,35 +126,25 @@ def regular_postseason_with_percentage(season_id)
...
-    reg_season_win_percent = percentages[1]
-    reg_season_win_percent.reduce({}) do |result, percent|
-      if !post_season_win_percent[percent[0]].nil?
-        result[percent[0]] = percent[1] - post_season_win_percent[percent[0]]
-      else
-        result[percent[0]] = percent[1]
-      end
+    post_season_perc, reg_season_perc = percentages[0], percentages[1]
+    reg_season_perc.reduce({}) do |result, perc|
+      !post_season_perc[perc[0]].nil? ? result[perc[0]] = perc[1] - post_season_perc[perc[0]] : result[perc[0]] = perc[1]
```
Comment (AlexisDumortier) [inline lib/season.rb]: we might want to change that one back to what it was if it limits readability

File: lib/season.rb [hunk 3]
```diff
@@ -106,16 +94,8 @@ def games_per_season_type(games, season_type)
...
-        result[game.away_team_id.to_s] = [game]
-      else
-        result[game.away_team_id.to_s] << game
-      end
-      if result[game.home_team_id.to_s].nil?
-        result[game.home_team_id.to_s] = [game]
-      else
-        result[game.home_team_id.to_s] << game
-      end
+    result[game.away_team_id.to_s].nil? ? result[game.away_team_id.to_s] = [game] : result[game.away_team_id.to_s] << game
```
Comment (AlexisDumortier) [inline lib/season.rb]: line is a bit long with the ternary operator (?)

File: lib/season.rb [hunk 4]
```diff
@@ -27,37 +25,29 @@ def games_per_coach_for_season(season_id)
...
       result
     end
   end
 
   def perc_per_coach(season_id)
     results_per_coach(season_id).reduce({}) do |result, coach_count|
-      result[coach_count[0]] = [0.0, 0.0] if result[coach_count[0]].nil?
-      result[coach_count[0]][0] = (coach_count[1][0] / (coach_count[1][0] + coach_count[1][1] + coach_count[1][2]).to_f).round(4)
-      result[coach_count[0]][1] = (coach_count[1][1] / (coach_count[1][0] + coach_count[1][1] + coach_count[1][2]).to_f).round(4)
+      result[coach_count[0]] = (coach_count[1][0] /  coach_count[1][1].to_f).round(4)
```
Comment (AlexisDumortier) [inline lib/season.rb]: removed the calculations that were not used by the method

File: lib/season.rb [hunk 5]
```diff
@@ -27,37 +25,29 @@ def games_per_coach_for_season(season_id)
   def results_per_coach(season_id)
     games_per_coach_for_season(season_id).reduce({}) do |result, coach_data|
       win_count = coach_data[1].count {|game| game.result == 'WIN'}
-      loss_count = coach_data[1].count {|game| game.result == 'LOSS'}
-      tie_count = coach_data[1].count {|game| game.result == 'TIE'}
-      result[coach_data[0]] = [win_count, loss_count, tie_count]
+      result[coach_data[0]] = [win_count, coach_data[1].size]
```
Comment (AlexisDumortier) [inline lib/season.rb]: removed unused calculations here

File: lib/season.rb (hunk 1 above)
Comment (palworth) [reply lib/season.rb]: Good use of syntax to shorten this method up

File: lib/season.rb (hunk 4 above)
Comment (palworth) [reply lib/season.rb]: Great refactor

**Comment by palworth** [comment]
Refactoring season class methods. Passing minitest and spec harness, ready to merge to master

**Pull request merged**